qiskit/providers/ibmq/credentials/hubgroupproject.py

```diff
@@ -14,7 +14,97 @@
 
 """Model for representing a hub/group/project tuple."""
 
-from collections import namedtuple
+from typing import Tuple, Optional
 
-HubGroupProject = namedtuple('HubGroupProject',
-                             ['hub', 'group', 'project'])
+from .exceptions import HubGroupProjectInvalidStateError
+
+
+class HubGroupProject:
+    """Class for representing a hub/group/project."""
+
+    def __init__(
+            self,
+            hub: str = None,
+            group: str = None,
+            project: str = None
+    ) -> None:
+        """HubGroupProject constructor."""
+        self.hub = hub
+        self.group = group
+        self.project = project
+
+    @classmethod
+    def from_str(cls, hgp: str) -> 'HubGroupProject':
```

Comment:
`from_stored_format()` would be a better name since it's more accurate and mirrors `to_stored_format()`.